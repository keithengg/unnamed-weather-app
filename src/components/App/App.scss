@tailwind base;
@tailwind components;
@tailwind utilities;

.main {
    display: flex;
    min-height: 100vh;
    height: 100%;
    width: 100vw;
    justify-content: center;
    background-size: cover;
    background-image: url(../../assets/bg-light.png);

    &--dark {
        background-image: url(../../assets/bg-dark.png);
    }

    &__container {
        display: flex;
        flex-direction: column;
        padding: 16px 32px 0px;
        width: 100%;
        max-width: 740px;
    }
}

.body__container,
.history__container {
    background-color: rgba(355, 355, 355, 0.3);
    border-radius: 15px;
    border: 1px solid #f2f2f2;
    padding: 32px;
}

.body__container {
    margin-top: 128px;
    height: 100%;
}

.history__container {
    margin: 16px 0px 16px;
    padding: 24px;
}