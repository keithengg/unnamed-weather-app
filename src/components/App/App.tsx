import React, { useState } from "react";
import Search from '../Search/Search';
import { getWeather } from "../../services/OpenWeatherMap";

import './App.scss';

const App: React.FC = () => {
	const [location, setLocation] = useState("")
	const [_, setWeatherData] = useState() // Opted to use useState and simple prop drilling as application is not deeply nested

	const handleUpdate = (location: string) => {
		setLocation(location)
	}

	const handleClick = () => {
		if (!location) return;

		getWeather(location)
			.then(data => setWeatherData(data))
			.catch(error => {
				console.error(error)
				return error
			})
	}

	return (
		<div>
			<Search location={location} handleUpdate={handleUpdate} handleClick={handleClick} />
		</div>
	);
}

export default App;
