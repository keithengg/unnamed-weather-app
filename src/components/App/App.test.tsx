import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("<App />", () => {
	it('should render correctly and match snapshot', () => {
		const { container } = render(<App />)
		expect(container).toMatchSnapshot()
	});
});