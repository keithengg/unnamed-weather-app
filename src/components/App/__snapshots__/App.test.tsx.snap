// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly and match snapshot 1`] = `
<div>
  <div
    class="main "
  >
    <div
      class="main__container"
    >
      <div
        class="flex"
      >
        <div
          class="search__container"
        >
          <label
            class="search__label "
            for="location"
          >
            Enter Country/City
          </label>
          <input
            class="search__input"
            name="location"
            type="text"
            value=""
          />
        </div>
        <button
          aria-label="Search"
          class="search__button"
          type="button"
        >
          <svg
            aria-hidden="true"
            aria-label="Search Icon"
            color="white"
            fill="none"
            height="32"
            stroke="currentColor"
            stroke-width="1.5"
            style="color: white;"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        class="body__container"
      >
        <div
          class="history__container"
        >
          <div>
            Search History
            <ul
              class="search-history__list"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
